test/e2e/network/service.go

```diff
@@ -4226,7 +4227,81 @@ var _ = common.SIGDescribe("Services", func() {
 		service, err := t.CreateService(service)
 		framework.ExpectNoError(err)
 
-		checkServiceReachabilityFromExecPod(ctx, f.ClientSet, ns, service.Name, service.Spec.ClusterIP, port)
+		_ = checkServiceReachabilityFromExecPod(ctx, f.ClientSet, ns, service.Name, service.Spec.ClusterIP, port)
+	})
+
+	ginkgo.It("should connect to the named ports during port number changing", func(ctx context.Context) {
+		serviceName := "mutable-named-port"
+		ns := f.Namespace.Name
+
+		t := NewServerTest(cs, ns, serviceName)
+		defer func() {
+			defer ginkgo.GinkgoRecover()
+			errs := t.Cleanup()
+			if len(errs) != 0 {
+				framework.Failf("errors in cleanup: %v", errs)
+			}
+		}()
+
+		port := int32(8080)
```

Comment:
It's probably best to call this `oldPort` or `origPort`, and then you can also define `newPort` (or `updatedPort` or whatever) here as well. And then later when you use them in a `ContainerPort`, it will be completely clear whether you're using the old or new value.